### PR TITLE
Cursor chaos

### PR DESCRIPTION
**Pull request by robojumper**

### CODE REVIEW

**Comment by robojumper** [comment]
Marking this as ready because There Was An Attempt (59%) and the splits and symbols are nice, but all the vtables and involved classes are super annoying and I can't be bothered to untangle them

File: src/d/d_cs_game.cpp [hunk 1]
```diff
@@ -0,0 +1,627 @@
...
+    {    "itemCursor_00_arrowPeak.brlan", "G_arrowCursor_00"},
+    {      "itemCursor_00_slingOn.brlan", "G_slingEffect_00"},
+    {    "itemCursor_00_slingKeep.brlan",  "G_slingGauge_00"},
+    {    "itemCursor_00_slingPeak.brlan",    "G_slingAll_00"},
+    {"itemCursor_00_slingPeakRing.brlan",   "G_slingRing_00"},
+    {     "itemCursor_00_slingOut.brlan", "G_slingEffect_00"},
+    {        "itemCursor_00_loop2.brlan",      "G_arrowLoop"},
+    {      "itemCursor_00_loopEnd.brlan",     "G_loopEnd_00"},
+    {     "itemCursor_00_loopEnd2.brlan",     "G_loopEnd_01"},
+};
```
Comment (elijah-thomas774) [inline src/d/d_cs_game.cpp:226]: Suggestion on moving this closer to the defines that act as an index into this?

File: include/d/d_cs_game.h [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+
+        EffectsStruct mEffects;
+        /* 0x070 */ dCsGame_c *mpCsGame;
+        EffectRelated mEffectRelated;
+        /* 0x0C8 */ m2d::ResAccIf_c *mpResAcc;
+        d2d::LytBase_c mLyt;
+        dCsGameAnmGroups_c mAnmGroups;
+        d2d::AnmGroups mAnm;
+        /* 0x??? */ u8 padding[0x44];
+        /* 0x72C */ lytBowCsr_c mBow;
```
Comment (elijah-thomas774) [inline include/d/d_cs_game.h:330]: Im starting to notice not commenting offsets. 
Should we keep commenting offsets or now due to clangd, should we not?

Comment (robojumper) [reply include/d/d_cs_game.h:330]: That was just me being lazy. I like hex offsets (outside of small RELs where they're probably not super necessary), and clangd only gives them in decimal so the hex offsets would still be nice here. I couldn't be bothered to math out the offsets in the inlined ctor, and I wasn't quite certain about which members belonged to classes with more inlined ctors and which members are standalone.

Comment (robojumper) [reply include/d/d_cs_game.h]: Added offsets, turns out I had one mistake around the EffectsStruct, which holds a dBase_c* or fBase_c* or void*, not a dAcBase_c*.

**Pull request merged**